include/aws/common/thread.h

```diff
@@ -37,20 +37,27 @@ typedef union {
 } aws_thread_once;
 #    define AWS_THREAD_ONCE_STATIC_INIT                                                                                \
         { NULL }
+typedef unsigned long aws_thread_id;
 #else
 typedef pthread_once_t aws_thread_once;
 #    define AWS_THREAD_ONCE_STATIC_INIT PTHREAD_ONCE_INIT
+typedef pthread_t aws_thread_id;
 #endif
 
+/*
+ * Number of chars to represent aws_thread_id as a string (2 hex chars per byte
+ * plus '\0' terminator). Needed for portable printing because pthread_t is
+ * opaque.
+ */
+#define AWS_THREAD_ID_REPR_LEN (sizeof(aws_thread_id) * 2 + 1)
```

Comment:
"length" when talking about c-strings doesn't usually include the null-terminator.
For example `AWS_DATE_TIME_STR_MAX_LEN` does not account for the null terminator.
Maybe `AWS_THREAD_ID_REPR_BUFSZ`? or keep it as `_LEN` but remove the +1?